### PR TITLE
Allow setting depths in BP hooks

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

Previously we selected "sensible" defaults for the hooks. It makes sense to expose the GraphResolveDepths to them so consumers can be opinionated if they want. This leads to substantial performance improvements for the frontend.

This work also discovered the bug fixed in #1530.

It also fixes bugs in the application where `Org`s weren't appearing in the workspace switcher.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203312852763953/1203458586282756/f) _(internal)_

## 🔍 What does this change?

See commit list please

## 📜 Does this require a change to the docs?

No

## ⚠️ Known issues

The Subgraph has started returning multiple versions of linked entities. We will fix this as part of the on-going versioning changes, but because the queries aren't actually temporal based at the moment (you can't ask for the graph at a given timestamp), it is easier to manually patch in a single place in the resolvers to maintain the invariant.

## 🛡 What tests cover this?

- None cover the specific code I've touched, to my knowledge

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Run the app
1.  Try the various routes..

## 📹 Demo
The org appearing again
image

### CODE REVIEW

File: packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/ontology-types-shim.ts [hunk 1]
```diff
@@ -39,7 +40,7 @@ export type OntologyCallbacks = {
 
 /* Data type CRU */
 export type AggregateDataTypesMessageCallback = MessageCallback<
-  {},
+  EmptyObject,
```
Comment (thehabbos007) [inline packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/ontology-types-shim.ts:43]: Shouldn't we allow specifying `constrainsValuesOn` here for non-primitive data types?

**Review by thehabbos007 — COMMENTED** [review]
I have some comments, but looks good otherwise. Happy to merge after these are addressed :)

File: packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts [hunk 2]
```diff
@@ -23,7 +28,26 @@ import { mapEntityModelToGQL } from "../model-mapping";
...
+ *   find the correct version of an entity.
+ *   https://app.asana.com/0/1201095311341924/1203331904553375/f
+ *
+ */
+const removeNonlatestEntities = (subgraph: Subgraph) => {
+  for (const entityId of Object.keys(subgraph.vertices)) {
+    if (isEntityId(entityId)) {
+      for (const oldVersion of Object.keys(subgraph.vertices[entityId]!)
+        .sort()
+        .slice(0, -1)) {
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts:44]: We're doing this to delete everything except the last element

File: packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/ontology-types-shim.ts (hunk 1 above)
Comment (Alfred-Mountfield) [reply packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/ontology-types-shim.ts:43]: They don't exist at the moment, so, no 😁 , there are tracked todo's elsewhere in the code to address this :)

File: packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts [hunk 3]
```diff
@@ -23,7 +28,26 @@ import { mapEntityModelToGQL } from "../model-mapping";
...
-/** @todo - rename these and remove "withMetadata" - https://app.asana.com/0/0/1203157172269854/f */
+/**
+ * @todo - Remove this when the Subgraph is appropriately queryable for a timestamp
+ *   at the moment, (not in the roots) all versions of linked entities are returned,
+ *   and with the lack of an `endTime`, this breaks the queryability of the graph to
+ *   find the correct version of an entity.
+ *   https://app.asana.com/0/1201095311341924/1203331904553375/f
+ *
+ */
+const removeNonlatestEntities = (subgraph: Subgraph) => {
```
Comment (benwerner01) [inline packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts:39]: Nit:
```suggestion
const removeNonLatestEntities = (subgraph: Subgraph) => {
```

Comment (Alfred-Mountfield) [reply packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts]: Thanks

**Review by thehabbos007 — APPROVED** [review]
I got `useCreateNewEntityAndRedirect` to work by commenting out some validation steps using WASM (currently there's an error going to http://localhost:3000/@alice/entities/new, that we've identified this earlier today on slack, https://hashintel.slack.com/archives/C022217GAHF/p1669787435717549). 

Creation & redirection is working fine if we get around the WASM error. Thank you!

**Pull request merged**